src/tools/compiletest/src/main.rs

```diff
@@ -211,11 +213,24 @@ pub fn parse_config(args: Vec<String>) -> Config {
         // Avoid spawning an external command when we know tidy won't be used.
         false
     };
+
+    let target_cfg = if cfg!(test) {
```

Comment:
I'm not sure this is great, but I don't know how else to get around the fact that a `rustc_path` seems not to be provided for test runs. There could be something obvious I'm missing though. I'm pretty tired.